src/HttpHeader.cc

```diff
@@ -356,15 +356,15 @@ HttpHeader::parse(const char *buf, size_t buf_len, bool atEnd, size_t &hdr_sz)
         blk_end = blk_start + strlen(blk_start);
     }
 
-    if (parse(blk_start, blk_end - blk_start)) {
+    if (parse(blk_start, blk_end - blk_start, code)) {
         hdr_sz = parse_start - buf;
         return 1;
     }
     return -1;
 }
 
 int
-HttpHeader::parse(const char *header_start, size_t hdrLen)
+HttpHeader::parse(const char *header_start, size_t hdrLen, const Http::StatusCode code)
```

Comment:
Pass a reference to a ContentLengthInterpreter object instead. Add a ContentLengthInterpreter::applyStatusCodeRules(sc) method that will result in the interpreter prohibiting (filtering out and ignoring) Content-Length headers as appropriate.

We should also add a ContentLengthInterpreter::applyRequestMethodRules(method) method for prohibiting Content-Length in GET/HEAD requests but I suspect that enhancement is outside this issue scope. Even if not added now, the need for applyRequestMethodRules() shows the symmetry across request/response parsing. See clientIsContentLengthValid().

Reply:
Done.